File: TopCoder01/PHP/Main.php
<?php
    class KiwiJuiceEasy {
    	public function thePouring($capacities, $bottles, $fromId, $toId)
		{
			for($i=0, $size = count($fromId);$i < $size; $i++){
				$sum = $bottles[$fromId[$i]] + $bottles[$toId[$i]];
				$bottles[$toId[$i]] = min($sum, $capacities[$toId[$i]]);
				$bottles[$fromId[$i]] = $sum - $bottles[$toId[$i]];
			}
			return $bottles;
		}
    }
	$capacities = array(30,20,10);
	$bottles = array(10,5,5);
	$fromId = array(0,1,2);
	$toId = array(1,2,0);
	$pouring = new KiwiJuiceEasy;
	$result = $pouring->thePouring($capacities, $bottles, $fromId, $toId);
	foreach ($result as $value) {
		echo "$value <br/>";
		
	}
?>

File: TopCoder01/Ruby/Main.rb
class KiwiJuiceEasy
  def thePouring(capacities, bottles, fromId, toId)
    for i in 0 ... fromId.length
      sum = bottles[fromId[i]] + bottles[toId[i]]
      bottles[toId[i]] = [sum, capacities[toId[i]]].min
      bottles[fromId[i]] = sum - bottles[toId[i]]
    end
    return bottles
  end
end

capacities = [30,20,10]
bottles = [10,5,5]
fromId = [0,1,2]
toId = [1,2,0]
puts KiwiJuiceEasy.new.thePouring(capacities, bottles, fromId, toId)